From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 10 May 2010 18:51:21 -0400
Subject: [net] netxen: compat header cleanup
Message-id: <20100510184905.2781.15777.sendpatchset@prarit.bos.redhat.com>
Patchwork-id: 24970
O-Subject: [RHEL5 PATCH 10/27] compat.h cleanup: netxen driver changes
Bugzilla: 546740
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

>From ba013bc4e4ae3ce268929d974604f65e2b028261 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 10 May 2010 09:32:22 -0400
Subject: [PATCH] netxen driver changes

Resolves 546740.

diff --git a/drivers/net/netxen/netxen_nic.h b/drivers/net/netxen/netxen_nic.h
index ae92c79..eb4b2e3 100644
--- a/drivers/net/netxen/netxen_nic.h
+++ b/drivers/net/netxen/netxen_nic.h
@@ -51,7 +51,6 @@
 
 #include "netxen_nic_hdr.h"
 #include "netxen_nic_hw.h"
-#include "netxen_nic_compat.h"
 
 #define _NETXEN_NIC_LINUX_MAJOR 4
 #define _NETXEN_NIC_LINUX_MINOR 0
diff --git a/drivers/net/netxen/netxen_nic_compat.h b/drivers/net/netxen/netxen_nic_compat.h
deleted file mode 100644
index 54553ab..0000000
--- a/drivers/net/netxen/netxen_nic_compat.h
+++ /dev/null
@@ -1,37 +0,0 @@
-#ifndef _NETXEN_NIC_COMPAT_H
-#define _NETXEN_NIC_COMPAT_H
-
-#ifndef list_splice_tail_init
-#define list_splice_tail_init list_splice_init
-#endif
-
-#ifndef vlan_dev_real_dev
-#define vlan_dev_real_dev(dev) VLAN_DEV_INFO(dev)->real_dev
-#endif
-
-#ifndef NETIF_F_LRO
-#define	NETIF_F_LRO		32768		/* large receive offload */
-#endif
-
-#ifndef work_func_t
-typedef void (*work_func_t)(void *);
-#endif
-
-#ifndef pr_fmt
-#define pr_fmt(fmt) fmt
-#endif
-
-#ifndef pr_err
-#define pr_err(fmt, ...) \
-	printk(KERN_ERR pr_fmt(fmt), ##__VA_ARGS__)
-#endif
-#ifndef pr_warning
-#define pr_warning(fmt, ...) \
-	printk(KERN_WARNING pr_fmt(fmt), ##__VA_ARGS__)
-#endif
-
-#ifndef to_net_dev
-#define to_net_dev(class) container_of(class, struct net_device, class_dev)
-#endif
-
-#endif
diff --git a/drivers/net/netxen/netxen_nic_hw.c b/drivers/net/netxen/netxen_nic_hw.c
index e241e8f..457040a 100644
--- a/drivers/net/netxen/netxen_nic_hw.c
+++ b/drivers/net/netxen/netxen_nic_hw.c
@@ -47,21 +47,6 @@ static void netxen_nic_io_write_128M(struct netxen_adapter *adapter,
 static u32 netxen_nic_io_read_128M(struct netxen_adapter *adapter,
 		void __iomem *addr);
 
-#ifndef readq
-static inline u64 readq(void __iomem *addr)
-{
-	return readl(addr) | (((u64) readl(addr + 4)) << 32LL);
-}
-#endif
-
-#ifndef writeq
-static inline void writeq(u64 val, void __iomem *addr)
-{
-	writel(((u32) (val)), (addr));
-	writel(((u32) (val >> 32)), (addr + 4));
-}
-#endif
-
 #define ADDR_IN_RANGE(addr, low, high)	\
 	(((addr) < (high)) && ((addr) >= (low)))
 
@@ -691,7 +676,7 @@ void netxen_p3_nic_set_multi(struct net_device *netdev)
 	struct list_head *head;
 	nx_mac_list_t *cur;
 
-	list_splice_tail_init(&adapter->mac_list, &del_list);
+	list_splice_init(&adapter->mac_list, &del_list);
 
 	nx_p3_nic_add_mac(adapter, adapter->mac_addr, &del_list);
 	nx_p3_nic_add_mac(adapter, bcast_addr, &del_list);
diff --git a/drivers/net/netxen/netxen_nic_init.c b/drivers/net/netxen/netxen_nic_init.c
index 6bff572..d56c4b8 100644
--- a/drivers/net/netxen/netxen_nic_init.c
+++ b/drivers/net/netxen/netxen_nic_init.c
@@ -1473,7 +1473,7 @@ netxen_process_lro(struct netxen_adapter *adapter,
 }
 
 #define netxen_merge_rx_buffers(list, head) \
-	do { list_splice_tail_init(list, head); } while (0);
+	do { list_splice_init(list, head); } while (0);
 
 int
 netxen_process_rcv_ring(struct nx_host_sds_ring *sds_ring, int max)
diff --git a/drivers/net/netxen/netxen_nic_main.c b/drivers/net/netxen/netxen_nic_main.c
index 7e15943..5a9018f 100644
--- a/drivers/net/netxen/netxen_nic_main.c
+++ b/drivers/net/netxen/netxen_nic_main.c
@@ -1238,9 +1238,9 @@ netxen_nic_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	pci_read_config_byte(pdev, PCI_REVISION_ID, &revision_id);
 
 	if ((revision_id >= NX_P3_A0) && (revision_id < NX_P3_B1)) {
-		pr_warning("%s: chip revisions between 0x%x-0x%x"
-				"will not be enabled.\n",
-				module_name(THIS_MODULE), NX_P3_A0, NX_P3_B1);
+		printk(KERN_WARNING "%s: chip revisions between 0x%x-0x%x"
+		       "will not be enabled.\n",
+		       module_name(THIS_MODULE), NX_P3_A0, NX_P3_B1);
 		return -ENODEV;
 	}
 
@@ -2686,7 +2686,7 @@ recheck:
 		goto done;
 
 	if (dev->priv_flags & IFF_802_1Q_VLAN) {
-		dev = vlan_dev_real_dev(dev);
+		dev = VLAN_DEV_INFO(dev)->real_dev;
 		goto recheck;
 	}
 
@@ -2722,7 +2722,7 @@ recheck:
 		goto done;
 
 	if (dev->priv_flags & IFF_802_1Q_VLAN) {
-		dev = vlan_dev_real_dev(dev);
+		dev = VLAN_DEV_INFO(dev)->real_dev;
 		goto recheck;
 	}
 
